File: EaseMall-biz/src/main/java/com/ysmull/easemall/annotation/Privilege.java
package com.ysmull.easemall.annotation;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

import static com.ysmull.easemall.model.entity.User.ROLE;

@Target(ElementType.METHOD)
@Retention(RetentionPolicy.RUNTIME)
public @interface Privilege {

    ROLE[] role() default {};

    boolean login() default false;
}

File: EaseMall-biz/src/main/java/com/ysmull/easemall/annotation/Privilege.kt
package com.ysmull.easemall.annotation

import com.ysmull.easemall.model.entity.User.ROLE
import kotlin.annotation.Retention
import kotlin.annotation.AnnotationRetention

@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
@Retention(AnnotationRetention.RUNTIME)
annotation class Privilege(val role: Array<ROLE> = [], val login: Boolean = false)
